### DISCUSSION TITLE
Reading (and Writing) of fields

### DISCUSSION THREAD

**Discussion by edsub** [john30/ebusd#1018] [Q&A] [answered]
I have succeeded in reading fields that are broadcasted by default as well.
```
$ ebusctl read -c vwzio Status01
39.0;-;18.836;-;50.5;off
```

But I cannot get reading of fields that are not broadcasted by default, that have 'no data stored'in find:
```
$ ebusctl find HwcTempDesired
ctlv2 HwcTempDesired = no data stored
```

```
$ ebusctl read -c ctlv2 HwcTempDesired
ERR: read timeout
```

How should I read these fields?

I have not changed my csv's
My ebusd startup parameters are:
`EBUSD_OPTS=" --scanconfig=full --configpath=/home/pi/ebusd.cfg --device=ens:/dev/ttyACM-EBUS  --httpport=8082 --port=8888 --enablehex --enabledefine --pollinterval=1 --receivetimeout=100 --sendretries=3 --accesslevel=* --latency=4 --loglevel=debug"` 

Obviously: Once I have this solved, the next challenge is to write values. Currentl an attempt leads to a read (?) timeout as well:
```
$ ebusctl write -c ctlv2 HwcTempDesired 34
ERR: read timeout

```
@john30 As the 'master' behind all of this you probably have a clue that I did not find in the Wiki ;-)

**Comment by edsub** [comment, marked as answer]
Well, all-in-all this took me a few evenings to figure out, but I think my issue is solved...by apply-ing the 20230917 Firmware to the ebusd v5 adapter (if this would not have solved my issue I was expecting to redo the ebus wiring and or have my installation checked by a technician)
There is nothing in the releasenotes (at least not in my interpretation ;-) ) that indicates this (or any previous) firmware should solve an issue like the one I had (hence why I did not try earlier and sticked with the first firmware from may).

So.... Finally I get a proper response to virtually (noticed some errors in CSV files, that is to be fixed separately) any read I now do,
e.g.
```
$ ebusctl read -f SystemFlowTemp
33

```

**Comment by john30** [comment]
cite: [fix potential bytes not sent to host via JTAG serial or UART or to eBUS](https://adapter.ebusd.eu/v5/ChangeLog#20230917)
